lib/SieveSemantics.php

```diff
@@ -379,10 +382,20 @@ protected function invoke_($token, $func, $arg = array())
             throw new SieveException($token, $err);
     }
 
-    protected function setRequire_($extension)
+    /**
+     * Add a require extension.
+     *
+     * @param string $extension the extension name.
+     * @return string|null an error message.
+     */
+    protected function setRequire_(string $extension)
     {
         array_push(self::$requiredExtensions_, $extension);
-        $this->registry_->activate($extension);
+        try {
+            $this->registry_->activate($extension);
+        } catch (\Throwable $throwable) {
+            return $throwable->getMessage();
```

Comment:
Where do we handle this message?

Reply:
In the function right above: https://github.com/ProtonMail/libsieve-php/blob/03ee689e6d24544302442510a7ce2da5d2690045/lib/SieveSemantics.php#L379